File: CodeForces/1501-1600/1512/A/1512A.cs
// Problem 1512A: Spy Detected!
// https://codeforces.com/contest/1512/problem/A
// Compiled with: Mono JIT compiler version 6.8.0.105 (Debian 6.8.0.105+dfsg-3.3+deb12u1 Sat Jun 21 16:33:59 UTC 2025)
// Linux version: Linux kernel 6.1.0-10-amd64
// Submitted on: October 23rd, 2025
// Codeforces language used: C# Mono 6.8

using System;

class Program {
    static void Main() {
        int runs = int.Parse(Console.ReadLine());
        for (int run = 0; run < runs; run++){
            int n = int.Parse(Console.ReadLine());
            string all_input = Console.ReadLine();
            string[] parts = all_input.Split(' ');

            // A case where the spy is at the start of the array
            if (parts[0] != parts[1]) {
                // If zero and two match, but zero and one don't one must be the spy
                // Otherwise, zero must be the spy if it doesn't match 0 or 1
                if (parts[0] == parts[2]) {
                    // Print 2 here because the problem expects 1-indexed answers
                    Console.WriteLine(2);
                } else {
                    Console.WriteLine(1);
                }
                continue;
            }
            // If the above case is false, we know parts[0] cannot be the spy, so we just need to reference everything to parts[0]
            for (int i = 0; i < n; i++){
                if (parts[0] != parts[i]) {
                    Console.WriteLine(i+1);
                    break;
                }
            }
        }
    }
}

File: CodeForces/1501-1600/1512/A/1512A.ts
// Problem 1512A: Spy Detected!
// https://codeforces.com/contest/1512/problem/A
// Compiled with: npx 9.2.0, tsc Version 5.9.3
// Executed with: node v20.19.4
// Linux version: Linux kernel 6.17.0-5-generic
// Written on: October 26th, 2025
// Codeforces language used: Cannot be submitted

import * as readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

async function main() {
    const rl = readline.createInterface({ input, output });
    
    const runs: number = parseInt((await rl.question('')).trim(), 10);

    for (let run = 0; run < runs; run++) {
        const number_count: number = parseInt((await rl.question('')).trim(), 10);
        const input_nums: number[] = (await rl.question('')).trim().split(' ').map((s: string) => parseInt(s, 10));

        // A case where the spy is either 0 OR 1
        if (input_nums[0] != input_nums[1]) {
            if (input_nums[0] != input_nums[2]) {
                console.log(1);
                continue;
            } else {
                console.log(2);
                continue;
            }
        }
        // We know for a fact that index 0 is not the spy if it equals 1
        const safe_value = input_nums[0];
        for (let i = 1; i < number_count; i++) {
            if (input_nums[i] != safe_value) {
                console.log(i+1);
                break;
            }
        }
    }
    rl.close();
}

main();
